tfjs-backend-webgl/src/backend_webgl.ts

```diff
@@ -176,6 +176,39 @@ export class MathBackendWebGL extends KernelBackend {
     return this.texData.numDataIds() - this.pendingDeletes;
   }
 
+  // Writes a new entry to the data store with a WebGL texture, and registers it
+  // to the texture manager.
+  writeTexture(
+      texture: WebGLTexture, shape: number[], dtype: DataType,
+      texHeight: number, texWidth: number, color: string): DataId {
+    // Temporarily create an tensor info to make the texture compatible with
+    // the runWebGLProgram's input.
+    const input = this.makeTensorInfo(shape, dtype);
+    const inData = this.texData.get(input.dataId);
+    // Even though the inpt texture could be unpacked or dense packed, it is
+    // always considered as unpacked for EncodeMatrixProgram and
+    // UnaryOpProgram.
+    inData.isPacked = false;
+
+    // Bind texture to the input tensor.
+    inData.texture = {texture, texShape: [texHeight, texWidth]};
+    inData.texShape = [texHeight, texWidth];
+
+    const shapeAs3D = webgl_util.getShapeAs3D(shape);
+    const program =
+        new EncodeMatrixProgram(shapeAs3D, false /* isByteArray */, color);
```

Comment:
Do we need to provide a special path for `RGBA` by calling `EncodeMatrixPackedProgram`? In this case, we can avoid once `runWebGLProgram` to pack the tensor (`PackProgram`) in many situations.

Reply:
RGBA input's packing schema is dense packing, while our tfjs's packing schema is square packing. As a result, even though both are packed (using all RGBA channels), we may need to transform the values through `EncodeMatrixPackedProgram`.